### PR TITLE
feat(new/resource): basic first resource

### PR DESCRIPTION
**Pull request by PsychoShock**
This pull request enhances the documentation for the first resource by expanding coverage, improving organization, and clarifying terminology. Additionally, it includes an extra section at the end of the module with supplementary materials and further reading suggestions for deeper learning.

### CODE REVIEW

File: content/docs/scripting-manual/introduction/creating-your-first-script-extra.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+weight: 490
+---
+
+Creating an "extra" or advanced section for your FiveM resource development guide is an excellent way to cater to developers who are ready to take their skills to the next level. This advanced section can delve deeper into topics such as security measures, optimization techniques, and advanced scripting practices. Here's how you can set up this advanced section in a new file:
+
+---
+
+# Advanced Topics in FiveM Resource Development
+
+Congratulations on making your first FiveM resource! As you continue your journey in FiveM development, you'll encounter more complex scenarios and challenges. This advanced guide aims to equip you with the knowledge and techniques to tackle these situations, ensuring your resources are robust, optimized, and secure.
```
Comment (Demigod916) [inline content/docs/scripting-manual/introduction/creating-your-first-script-extra.md:12]: Thank you chat gippity. You're my best friend.

File: content/docs/scripting-manual/introduction/creating-your-first-script-extra.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+---
+title: How to create your first FiveM resource (advanced part)
+weight: 490
+---
+
+Creating an "extra" or advanced section for your FiveM resource development guide is an excellent way to cater to developers who are ready to take their skills to the next level. This advanced section can delve deeper into topics such as security measures, optimization techniques, and advanced scripting practices. Here's how you can set up this advanced section in a new file:
```
Comment (Demigod916) [inline content/docs/scripting-manual/introduction/creating-your-first-script-extra.md:6]: This is the part you are supposed to delete

**Comment by ESK0** [comment]
Imagine deleting true comments :) I really didnt expect that... Now lets create whole docs by LLM

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 3]
```diff
@@ -1,283 +1,265 @@
...
-    -- and force respawn when the game type starts
-    exports.spawnmanager:forceRespawn()
-end)
+shared_script 'shared/main_sh.lua'
+client_script 'client/main_cl.lua'
+server_script 'server/main_sv.lua'
 ```
 
-A quick mention of the difference between client and server scripts: most of what you'll do in FiveM will be done using client scripts, since in current versions there's no interaction with _game_ functionality in server scripts. Server scripts should be used to have scripted actions occur _across_ clients (using client/server events), and to provide a 'source of trust' for various actions, such as storing/loading things in a persistent database.
+Copy this code and past it in the `fxmnifest.lua` file. It should look like this after:
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:67]: Typo `fxmnifest.lua`

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 4]
```diff
@@ -1,283 +1,265 @@
...
+    -- Trigger a client event with the value
+    TriggerClientEvent('cfx:client:firstEvent', source, value)
+end, false)
+```
 
-It's silly to close your game and server and restart them both to iterate on your resource. Of course, you can _restart_ your resource as well.
+Let's break this in part:
+#### 1. Checking if Code is Executed by a Player:
+```lua
+if not (source > 0) then return end
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:105]: `if source ~= 0 then return end`

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 5]
```diff
@@ -1,283 +1,265 @@
...
+The manifest will read the following:
+- Shared script in the `shared` folder
+- Client script in the `client` folder
+- Server script in the `server` folder
 
-You're probably hoping to be able to run this little example - well, hopefully you already have a running FXServer instance - if not, follow the guide for that.
+**Note:** You're free to name the folders/files whatever you wish. The names provided here are just common conventions.
 
-Once you've started FXServer, execute the `refresh` command in the console. This'll reread every single `fxmanifest.lua` file for every resource you have installed, since you probably just started the server this isn't _really_ needed but if you had the server running already this is just A Good Idea™ to do.
+## Server resource
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:78]: Server file

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 6]
```diff
@@ -1,283 +1,265 @@
...
 
-You'll probably want to do more. For this, you're going to have to learn how to _call natives_, which has nothing to do with indigenous people and actually are a R* label for 'game-defined script functions'. There's a lot of intricacies involved in calling natives properly - for a full reference, see the special section for this - but we'll start simple for now.
+#### Preview
+After following every steps of the server section. The code inside `main_sv` should look like this.
 
-In a stupid way of 'this trope again', we'll make a command that'll spawn a car. Locally. Because nobody cares about the server when they're starting out.
+<img src="img/example_server_code.png" alt="pic" width="500">
 
-At the bottom of your `mymode_client.lua`, add this code:
+## Client resource
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:136]: Client file

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 7]
```diff
@@ -1,283 +1,265 @@
...
-    })
-end, false)
-```
+To continue the guide, we will create another file `main_cl.lua` inside `client` folder. This should look like this:
 
-Starting already, we see a call to a function. We did not define that function. Well, _we_ (as in, the FiveM team) did, but not when guiding you, the reader, through this wondrously written marvel of a guide. That means it must come from somewhere else!
+<img src="img/resource_folder_client.png" alt="pic" width="500">
 
-And, guess what, it's actually {{% native_link "REGISTER_COMMAND" %}}! Click that link, and you'll be led to the documentation for this native. It looks a bit like this:
+The client-side files, located in the `client` folder, control interactions visible to players and handle rendering, while also enforcing security measures and processing that cannot be tampered with by players.
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:142]: Client *can* be tampered with.

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 8]
```diff
@@ -1,283 +1,265 @@
...
 
-We'll mainly care about the name on the second line (`RegisterCommand`, as used in the Lua code above), and the arguments.
-
-As you can see, the first argument is the **command name**. The second argument is a **function** that is the command handler, and the third argument is a **boolean** that specifies whether or not it should be a restricted command.
+Let's break this in part:
+#### 1. Checking if param is valid
+```lua
+if not information then return end
+```
+This line serves as a crucial validation step, ensuring that the code receives a valid parameter sent by the server. Detecting and handling invalid parameters can be instrumental in identifying and thwarting cheating attempts. For further insights on advanced techniques, refer to the [advanced part](/content/docs/scripting-manual/introduction/creating-your-first-script-extra.md).
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:160]: Why would a value sent from the server help in identifying cheating attempts?

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 9]
```diff
@@ -1,283 +1,265 @@
...
-    SetPedIntoVehicle(playerPed, vehicle, -1)
+### Alternative Scripting Languages
 
-    -- give the vehicle back to the game (this'll make the game decide when to despawn the vehicle)
-    SetEntityAsNoLongerNeeded(vehicle)
+While Lua is a powerful and widely used scripting language in the FiveM community, it's not the only option available. FiveM supports several other scripting languages, allowing you to choose the one that best fits your project's needs and your personal preferences. Here are some alternatives you might consider:
 
-    -- release the model
-    SetModelAsNoLongerNeeded(vehicleName)
+- **JavaScript (with NPM)**: For developers familiar with JavaScript, FiveM supports JS scripting using the [NPM package](https://www.npmjs.com/package/fivem). This allows you to leverage the vast ecosystem of JavaScript libraries and tools, making it a compelling choice for certain projects.
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:228]: This package is not related to the JavaScript runtime at all, it's for getting server status.

File: content/docs/scripting-manual/introduction/creating-your-first-script.md [hunk 10]
```diff
@@ -1,283 +1,265 @@
...
 
-#### Step 3: Waiting for the model to be loaded
-We loop calls to {{% native_link "HAS_MODEL_LOADED" %}} to check if loading succeeded. Since this is a loop and we're cooperatively multitasked, you'll have to give the game time to run as well - otherwise it'll never even finish loading and the game will unfortunately freeze. That's what the `Wait` call is for - it waits for the specified amount of milliseconds, then returns right back into the script.
+- **FiveM Discord Server**: Joining the [FiveM Discord server](https://discord.gg/fivem) is one of the quickest ways to connect with the community. It's a hub for real-time discussions, with channels dedicated to scripting help, resource development, and server management.
 
-Once the model's loaded, we'll continue.
+- **CitizenFX Collective**: The [CitizenFX Collective](https://citizen.re/collective/) is a platform where you can discover and support projects within the FiveM community. It's a great way to find resources, tools, and even contribute to the development of FiveM itself.
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/creating-your-first-script.md:242]: ??

**Comment by AvarianKnight** [comment]
I don't think this is an improvement over the current documentation.

I added this to the [tracler](https://github.com/citizenfx/fivem-docs/issues/466) because in the future these scripting examples should be completely rewritten and should preferably be a step-by-step in making an example resource like capture the flag.

**Comment by PsychoShock** [comment]
> In the future these scripting examples should be completely rewritten and should preferably be a step-by-step in making an example resource like capture the flag.

Well this is **exactly** having a step-by-step doc. It also handles docs having beginer guide + more advanced. If it will be rewrittent in the future it's fine, but closing this instead of the actual one make no sense imo. This is a way more complete guide + up to date.
